### DISCUSSION TITLE
New archive system

### DISCUSSION THREAD

**Discussion by Tzahi12345** [Tzahi12345/YoutubeDL-Material#898] [General]
Wanted to open a discussion regarding the new archive system in v4.3.1. Basically, archive files are no more, and archives are handled internally within the database.

I probably should've made this a discussion before implementing this, but we had file system problems when it came to managing large archive files (e.g. #777, #755)

There's a couple things you need to know about this change:

## New archive manager

You can manage archives from the new archive manager by hitting the 3 dots in the top right corner. You can import and export archives from here. (I'll upload a screenshot here later)

## Subscriptions temporary archive

No archive files are generated from one-off files, but we do generate a temporary archive for subscriptions during the "check subscription for any missing videos" stage. No videos get added to the archive because it's not downloading anything, but it speeds things up during the check as it knows to avoid extracting info (which is slow) for videos in the archive.

## Delete forever option for everyone!

Even with archives disabled, you can now delete a video forever in a subscription. This is functionality everyone expects, but isn't possible without either leveraging the archive system or by implementing it internally.

## Migration

We auto migrate your existing archives into the new system. Here's the relevant code for that:

https://github.com/Tzahi12345/YoutubeDL-Material/blob/e573f34cea93dedaa3268beaf50611648e04535f/backend/archive.js#L31-L52

Basically, any line that is mangled it will skip. Any line that has two strings separated by a space, it will interpret as `<extractor_id> <video id>`, for example `youtube dQw4w9WgXcQ`

Let me know if this doesn't work well for you.

--------------

That's pretty much it. Let me know y'alls thoughts and I'll try to include it in the next update
